services/director-v2/src/simcore_service_director_v2/modules/comp_scheduler/base_scheduler.py

```diff
@@ -246,88 +327,146 @@ async def _get_changed_tasks_from_backend(
+            *(
+                publish_service_resource_tracking_started(
+                    self.rabbitmq_client,
+                    service_run_id=get_resource_tracking_run_id(
+                        user_id, t.project_id, iteration
+                    ),
+                    wallet_id=run_metadata.get("wallet_id", -1),
+                    wallet_name=run_metadata.get("wallet_name", _UNDEFINED_METADATA),
+                    product_name=run_metadata.get("product_name", _UNDEFINED_METADATA),
+                    simcore_user_agent=run_metadata.get(
+                        "simcore_user_agent", UNDEFINED_DEFAULT_SIMCORE_USER_AGENT_VALUE
+                    ),
+                    user_id=user_id,
+                    user_email=run_metadata.get("user_email", _UNDEFINED_METADATA),
+                    project_id=t.project_id,
+                    project_name=run_metadata.get("project_name", _UNDEFINED_METADATA),
+                    node_id=t.node_id,
+                    node_name=run_metadata.get("node_id_names_map", {}).get(
+                        t.node_id, _UNDEFINED_METADATA
+                    ),
+                    service_key=ServiceKey(t.image.name),
+                    service_version=ServiceVersion(t.image.tag),
+                    service_type=ServiceType.COMPUTATIONAL,
+                    service_resources=create_service_resources_from_task(t),
+                    service_additional_metadata={},
+                )
+                for t in tasks
+            )
+        )
+        # instrumentation
+        await asyncio.gather(
+            *(
+                publish_service_started_metrics(
+                    self.rabbitmq_client,
+                    user_id=user_id,
+                    simcore_user_agent=run_metadata.get(
+                        "simcore_user_agent", UNDEFINED_DEFAULT_SIMCORE_USER_AGENT_VALUE
+                    ),
+                    task=t,
+                )
+                for t in tasks
+            )
+        )
+
+        # update DB
         comp_tasks_repo = CompTasksRepository(self.db_engine)
         await asyncio.gather(
             *(
                 comp_tasks_repo.update_project_tasks_state(
-                    t.project_id, [t.node_id], t.state
+                    t.project_id,
+                    [t.node_id],
+                    t.state,
+                    optional_started=utc_now,
                 )
                 for t in tasks
             )
         )
 
-    async def _publish_service_started_metrics(
-        self,
-        user_id: UserID,
-        project_id: ProjectID,
-        changed_tasks: list[tuple[_Previous, _Current]],
-    ) -> None:
-        for previous, current in changed_tasks:
-            if current.state is RunningState.STARTED or (
-                previous.state in WAITING_FOR_START_STATES
-                and current.state in COMPLETED_STATES
-            ):
-                message = InstrumentationRabbitMessage.construct(
```

Comment:
Could you please also replace the following calls in the following as well, since you are changing this:

https://github.com/ITISFoundation/osparc-simcore/blob/54970e3cb376515baa8e174db77d6c56be339b27/services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_events.py#L112

https://github.com/ITISFoundation/osparc-simcore/blob/54970e3cb376515baa8e174db77d6c56be339b27/services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_events_utils.py#L351

Reply:
not sure I follow what you want here?

Reply:
you mean using the construct instead of parsing what should not be parsed?